services/resource_dereference_service.go

```diff
@@ -0,0 +1,101 @@
+package services
+
+import (
+	// jsonpb Marshaller is deprecated, but is needed because there's only one way to proto
+	// marshal in combination with our proto generator version
+	"encoding/json"
+
+	"github.com/cheqd/did-resolver/types"
+	"github.com/cheqd/did-resolver/utils"
+)
+
+type ResourceDereferenceService struct {
+	ledgerService LedgerServiceI
+	didDocService DIDDocService
```

Comment:
May be not depend on didDoc service here? It doesn't seem like we need anything DidDoc related here. The service is used for content stream marshalling only which is pretty generic. What do you think about extracting this method into let's say `resolution_service` or helpers?